src/core/OmniFormElement.ts

```diff
@@ -0,0 +1,375 @@
+ *
+ * @cssprop --omni-form-focussed-border-width - Form focussed border width.
+ * @cssprop --omni-form-focussed-border-color - Form focussed border color.
+ * @cssprop --omni-form-focussed-label-color - Form focussed label color.
+ *
+ * @cssprop --omni-form-error-label-color - Form error label color.
+ * @cssprop --omni-form-error-border-color - Form error border color.
+ *
+ * @cssprop --omni-form-label-disabled-color - Form label disabled color.
+ * @cssprop --omni-form-disabled-border-color - Form disabled border color.
+ * @cssprop --omni-form-disabled-background-color - Form disabled background color.
+ * @cssprop --omni-label-disabled-focus-background-color - Form label disabled focus background color.
+ *
+ * @cssprop --omni-form-hint-label-font-color - Form hint label font color.
+ * @cssprop --omni-form-hint-label-font-family - Form hint label font family.
+ * @cssprop --omni-form-hint-label-font-size - Form hint label font size.
+ * @cssprop --omni-form-hint-label-font-weight - Form hint label font weight.
+ * @cssprop --omni-form-hint-label-padding-top - Form hint label top padding.
+ * @cssprop --omni-form-hint-label-padding-left - Form hint label left padding.
+ * @cssprop --omni-form-hint-label-border-width -
+ *
+ * @cssprop --omni-form-error-label-font-color - Form error label font color.
+ * @cssprop --omni-form-error-label-font-family - Form error label font family.
+ * @cssprop --omni-form-error-label-font-size - Form error label font size.
+ * @cssprop --omni-form-error-label-font-weight - Form error label font weight.
+ * @cssprop --omni-form-error-label-padding-top - Form error label top padding.
+ * @cssprop --omni-form-error-label-padding-left - Form error label left padding.
+ * @cssprop --omni-form-error-label-border-width
+ *
+ * @cssprop --omni-form-hover-color - Form hover color.
+ * @cssprop --omni-form-disabled-hover-color - Form disabled hover color.
+ * @cssprop --omni-form-error-hover-color - Form error hover color.
+ *
+ */
+export class OmniFormElement extends LitElement {
+    /**
+     * Text label.
+     * @attr
+     */
+    @property({ type: String, reflect: true }) label: string;
+
+    /**
+     * The value entered into the form.
+     * @attr
+     */
+    @property({ reflect: true }) value: string | number = null;
+
+    /**
+     * Data associated with the component.
+     * @attr
+     */
+    @property({ type: Object, reflect: true }) data: object;
+
+    /**
+     * A hint message to assist the user.
+     * @attr
+     */
+    @property({ type: String, reflect: true }) hint: string;
+
+    /**
+     * A error message guiding a user to correct a mistake.
+     * @attr
+     */
+    @property({ type: String, reflect: true }) error: string;
+
+    /**
+     * Indicator if the component should be disabled.
+     * @attr
+     */
+    @property({ type: Boolean, reflect: true }) disabled = false;
+
+    @query('.form-container')
+    private _formContainer: HTMLElement;
```

Comment:
Should this not be called _formContainerElement?

Reply:
Updated to suggested name.